### DISCUSSION TITLE
Problem with pysr.julia_helpers (because PyCall got messed up)

### DISCUSSION THREAD

**Discussion by unary-code** [MilesCranmer/PySR#368] [Q&A]
I'll share my full code "Use_the_PySR_Package" here.

Actually, interestingly I was able to successfully run these lines of code (including the last
```jl = pysr.julia_helpers.init_julia(julia_kwargs={"threads": 8, "optimize": 3})```
line of code) in a separate smaller new ipynb file called Use_just_the_PySR_Julia_Helper.ipynb .

I must have run some other command in this one Use_the_PySR_Package  ipynb file that messed up this one file permanently, because even when I re-started the Jupyter Notebook, I still got a error when running
```jl = pysr.julia_helpers.init_julia(julia_kwargs={"threads": 8, "optimize": 3})```.

========================
In the Use_the_PySR_Package ipynb file,
I ran
```
!curl -fsSL https://install.julialang.org | sh
```
which tells me that Julia is installed in this system
then
I ran
```
!pip3 install -U pysr
```
which tells me a bunch of "requirement already satisfied"'s, including "Requirement already satisfied: pysr in /opt/anaconda3/envs/pysr_env/lib/python3.10/site-packages (0.14.3)"
then I ran
```
!python3 -m pysr install
```
then I ran
```
import pysr
```
then I ran
```
jl = pysr.julia_helpers.init_julia(julia_kwargs={"threads": 8, "optimize": 3})
```
which gives me this error message:
```
JuliaError: Exception 'ArgumentError' occurred while calling julia code:
const PyCall = Base.require(Base.PkgId(Base.UUID("438e738f-606a-5dbb-bf0a-cddfbfd45ab0"), "PyCall"))
```

Any idea why?
My version of Julia is 1.9.1
My version of Python is 3.10.4

Notably, this error did not happen to me before. A few days ago, I was able to create the julia object
```
jl = pysr.julia_helpers.init_julia(julia_kwargs={"threads": 8, "optimize": 3})
```

and I was able to do
```
jl.eval("""
print("hello world")
""")
```
for example.

The only difference I can think of between a few days ago and today is that today, I imported the Roots and ForwardDiff package
```
jl.eval(
"Pkg.add("Roots")
Pkg.add("ForwardDiff")

using Roots
using ForwardDiff")
```
But I have a feeling there is some other reason for this, as I do not understand why doing that would cause a problem with using PyCall.

[Use_the_PySR_Package.pdf](https://github.com/MilesCranmer/PySR/files/11951998/Use_the_PySR_Package.pdf)
[Use_just_the_PySR_Julia_Helper.pdf](https://github.com/MilesCranmer/PySR/files/11951999/Use_just_the_PySR_Julia_Helper.pdf)

**Comment by MilesCranmer** [comment]
Hi @unary-code,

Thanks for the report. Could you edit your post to surround code with three back quotes `\`` so that it is formatted correctly? Otherwise it is hard to read.

Regarding the issue, the latest PySR uploaded today has some functionality to self-repair PyCall installs. Could you try upgrading your PySR and retrying?

Thanks,
Miles

**Reply by unary-code** [reply]
Oh okay sure, I just edited my original post to include the ``` back quotes.

I will upgrade my PySR version in my conda virtual environment, then re-install PySR and import PySR in the ipynb jupyter notebook. Then I'll briefly summarize what ends up happening (success or failure) on this thread.

**Reply by MilesCranmer** [reply]
Also; since you are using conda: are you using pip or conda to install PySR? It’s generally a bad idea to mix install methods as there might be two versions of packages
